src/reflect/value.go

```diff
@@ -217,7 +217,15 @@ func (v Value) CanAddr() bool {
 }
 
 func (v Value) Addr() Value {
-	panic("unimplemented: (reflect.Value).Addr()")
+	if !v.CanAddr() {
+		panic("taking addr of un-addr-able value")
+	}
+
+	return Value{
+		typecode: PtrTo(v.Type()).(*rawType),
```

Comment:
Interfaces can be expensive, so what I usually do is create a separate function that works directly with `*rawType`. Something like this:

```suggestion
		typecode: pointerTo(v.typecode),
```

```go
func pointerTo(t *rawType) *rawType { ... }
func PointerTo(t Type) Type {
    return pointerTo(t.(*rawType))
}
```